src/index.js

```diff
@@ -109,6 +109,13 @@ export class BrowserHistory extends History {
   }
 
   /**
+   * Returns the origin portion of the current history object.
+   */
+  getOrigin(): string {
+    return _createOrigin(this.location.protoco, this.location.hostname, this.location.port);
```

Comment:
Typo: `location.protoco`. We should test this method.

We will need to join the origin with `this.root` to handle cases where the app root is a subdirectory. I think that also means that `getRoot` or `getBaseUrl` would be a more appropriate name, since this method does not necessarily return an origin.